model/src/main/kotlin/HashAlgorithm.kt

```diff
@@ -115,7 +122,14 @@ enum class HashAlgorithm(private vararg val aliases: String, val verifiable: Boo
             // lot of memory at the same time, also considering that this function could potentially be run on multiple
             // threads in parallel.
             val buffer = ByteArray(4 * 1024 * 1024)
-            val digest = MessageDigest.getInstance(toString())
+            val digest = if (this == SHA1_GIT) {
```

Comment:
This `if` construct is a bit smelly, but I have no better idea either. I was thinking whether `HashAlgorithm` could define a function `getMessageDigest()` that could be overridden by `SHA1_GIT`, but as the initialization of the digest requires the file length, this is probably not possible in a generic way.

Reply:
> This `if` construct is a bit smelly

Why?

Reply:
In OO-style you typically try to avoid such constructs by using polymorphism.

Reply:
Kotlin encourages functional programming anyway 😉

Reply:
Let's say, it is hybrid. So we can use the best of both worlds ;-)

Reply:
You could add a `getDigest()` function to the class that defaults to `MessageDigenst.getInstance(toString())` and override it for `SHA1_GIT`.

Reply:
> I was thinking whether `HashAlgorithm` could define a function `getMessageDigest()` that could be overridden by `SHA1_GIT`, but as the initialization of the digest requires the file length, this is probably not possible in a generic way.

I'd say as this is the only function of the class it would be ok to pass the `file` to `getMessageDigest()` even though it's ignored by the default implementation.

Reply:
Done.